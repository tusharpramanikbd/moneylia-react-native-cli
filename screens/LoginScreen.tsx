import {
  Image,
  Pressable,
  SafeAreaView,
  StatusBar,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import React from 'react';
import type {NativeStackScreenProps} from '@react-navigation/native-stack';
import RootStackParamList from '../@types/RootStackPramList';

type Props = NativeStackScreenProps<RootStackParamList, 'Login'>;

// For Unit Testing
// type LoginScreenProp = {
//   navigation: any;
// };

const LoginScreen = ({navigation}: Props) => {
  const loginOnPressHandler = () => {
    navigation.navigate('BottomTabs');
  };

  return (
    <SafeAreaView style={styles.rootContainer}>
      <StatusBar barStyle="dark-content" backgroundColor="#FFFFFF" />
      <View style={styles.container}>
        <Image
          style={styles.image}
          source={require('../assets/images/login-image.png')}
        />
        <Text style={styles.titleText}>Save your money</Text>
        <Text style={styles.loremText}>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est in quis
          erat a sit.
        </Text>
        <View style={styles.buttonStyle}>
          <Pressable
            android_ripple={{color: '#dddddd'}}
            onPress={loginOnPressHandler}>
            <Text style={styles.buttonText}>Join for free</Text>
          </Pressable>
        </View>
        <Text style={styles.linkText}>
          Don't have SPID or CIE? Find out more
        </Text>
      </View>
    </SafeAreaView>
  );
};

export default LoginScreen;

const styles = StyleSheet.create({
  rootContainer: {
    flex: 1,
  },
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 40,
  },
  image: {
    width: 335,
    height: 334,
    marginBottom: 24,
  },
  titleText: {
    textAlign: 'center',
    fontSize: 32,
    fontWeight: 'bold',
    color: '#3A3B7B',
    marginBottom: 8,
  },
  loremText: {
    textAlign: 'center',
    marginBottom: 60,
    color: '#6F6D7B',
  },
  buttonStyle: {
    marginBottom: 24,
    width: '85%',
    backgroundColor: '#7476ED',
    borderRadius: 4,
    elevation: 4,
  },
  buttonText: {
    textAlign: 'center',
    color: '#FFFFFF',
    paddingVertical: 16,
    fontWeight: 'bold',
    fontSize: 18,
  },
  linkText: {
    color: '#3A3B7B',
    textDecorationLine: 'underline',
  },
});
