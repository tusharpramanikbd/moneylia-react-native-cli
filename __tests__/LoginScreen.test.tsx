import React from 'react';
import LoginScreen from '../screens/LoginScreen';
import {fireEvent, render} from '@testing-library/react-native';
import renderer from 'react-test-renderer';

describe('Login Screen', () => {
  it('should go to dashboard screen on login', () => {
    const navigation = {navigate: () => {}};
    spyOn(navigation, 'navigate');
    const page = render(<LoginScreen navigation={navigation} />);

    const loginButton = page.getByText('Join for free');
    fireEvent.press(loginButton);

    expect(navigation.navigate).toHaveBeenCalledWith('BottomTabs');
  });

  it('renders correctly', () => {
    const navigation = {navigate: () => {}};
    spyOn(navigation, 'navigate');
    const tree = renderer
      .create(<LoginScreen navigation={navigation} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
