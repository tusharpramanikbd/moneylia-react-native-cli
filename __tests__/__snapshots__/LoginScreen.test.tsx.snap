// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 40,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/login-image.png",
        }
      }
      style={
        Object {
          "height": 334,
          "marginBottom": 24,
          "width": 335,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#3A3B7B",
          "fontSize": 32,
          "fontWeight": "bold",
          "marginBottom": 8,
          "textAlign": "center",
        }
      }
    >
      Save your money
    </Text>
    <Text
      style={
        Object {
          "color": "#6F6D7B",
          "marginBottom": 60,
          "textAlign": "center",
        }
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est in quis erat a sit.
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#7476ED",
          "borderRadius": 4,
          "elevation": 4,
          "marginBottom": 24,
          "width": "85%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "bold",
              "paddingVertical": 16,
              "textAlign": "center",
            }
          }
        >
          Join for free
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#3A3B7B",
          "textDecorationLine": "underline",
        }
      }
    >
      Don't have SPID or CIE? Find out more
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`Login Screen should be visible save your money text 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 40,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/login-image.png",
        }
      }
      style={
        Object {
          "height": 334,
          "marginBottom": 24,
          "width": 335,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#3A3B7B",
          "fontSize": 32,
          "fontWeight": "bold",
          "marginBottom": 8,
          "textAlign": "center",
        }
      }
    >
      Save your money
    </Text>
    <Text
      style={
        Object {
          "color": "#6F6D7B",
          "marginBottom": 60,
          "textAlign": "center",
        }
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est in quis erat a sit.
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#7476ED",
          "borderRadius": 4,
          "elevation": 4,
          "marginBottom": 24,
          "width": "85%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "bold",
              "paddingVertical": 16,
              "textAlign": "center",
            }
          }
        >
          Join for free
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#3A3B7B",
          "textDecorationLine": "underline",
        }
      }
    >
      Don't have SPID or CIE? Find out more
    </Text>
  </View>
</RCTSafeAreaView>
`;
